src/main/java/de/tum/in/www1/artemis/service/SubmissionService.java

```diff
@@ -183,8 +183,8 @@ public void checkSubmissionLockLimit(long courseId) {
             participations = studentParticipationRepository.findByExerciseIdWithLatestSubmissionWithoutManualResults(exercise.getId());
         }
 
-        List<Submission> submissionsWithoutResult = participations.stream().map(StudentParticipation::findLatestSubmission).filter(Optional::isPresent).map(Optional::get)
-                .collect(Collectors.toList());
+        List<Submission> submissionsWithoutResult = participations.stream().map(participation -> participation.findLatestSubmission(true)).filter(Optional::isPresent)
```

Comment:
Maybe it would be an idea to use two methods instead of using a parameter?
So maybe `findLatestSubmission` and `findLatestLegalSubmission`. Then you wouldn't have to pass a parameter every time you only want the latest legal one (which will probably be the case most of the time)